Uplift/Views/TabbedView/TabbedViewController.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  File.swift
+//  Uplift
+//
+//  Created by alden lamp on 10/20/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+class TabbedViewController: UIViewController {
+
+    let tabbedControl: TabbedControl!
+    let viewControllers: [UIViewController]!
+
+    private var selectedTab = 0
+    private var slideViewLeading: ConstraintMakerEditable?
+
+    init(tabbedControl: TabbedControl, viewControllers: [UIViewController]) {
+        self.tabbedControl = tabbedControl
+        self.viewControllers = viewControllers
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupViews()
+        setupConstraints()
+    }
+
+    func setupViews() {
+        for viewController in viewControllers {
+            viewController.view.translatesAutoresizingMaskIntoConstraints = false
+            self.view.addSubview(viewController.view)
+        }
+
+        tabbedControl.translatesAutoresizingMaskIntoConstraints = false
+        tabbedControl.delegate = self
+        self.view.addSubview(tabbedControl)
+    }
+
+    func setupConstraints() {
+        tabbedControl.snp.updateConstraints { make in
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview()
+            make.top.equalToSuperview()
+            make.height.equalTo(45)
+        }
+
+        for (index, viewController) in viewControllers.enumerated() {
+            viewController.view.snp.makeConstraints { make in
+                make.top.equalTo(tabbedControl.snp.bottom)
+                make.bottom.equalToSuperview()
+                make.width.equalToSuperview()
+                if index == selectedTab {
+                    slideViewLeading = make.leading.equalTo(self.view.snp.leading)
+                }
+                if index != 0 {
```

Comment:
Is there ever a case where index != selectedTab or 0? Is it still constrained properly then?

Reply:
Yeah, its constraining views in a list to each other (even outside the window). Hence only having the selected tab constrainted to self.view but linking their leadings to trailings.